x/incentives/abci.go

```diff
@@ -0,0 +1,36 @@
+package incentives
+
+import (
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/c-osmosis/osmosis/x/incentives/keeper"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// BeginBlocker is called on every block
+func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k keeper.Keeper) {
+}
+
+// Called every block to distribute coins
+func EndBlocker(ctx sdk.Context, k keeper.Keeper) []abci.ValidatorUpdate {
+	currentEpoch, epochBeginBlock := k.GetCurrentEpochInfo(ctx)
+	params := k.GetParams(ctx)
+
+	if ctx.BlockHeight() < epochBeginBlock+params.BlocksPerEpoch { // not time for epoch
+		return []abci.ValidatorUpdate{}
+	}
+
+	// update epoch info
+	k.SetCurrentEpochInfo(ctx, currentEpoch+1, ctx.BlockHeight())
+
+	// distribute due to epoch event
+	pots := k.GetActivePots(ctx)
+	for _, pot := range pots {
+		k.Distribute(ctx, pot)
+		if pot.NumEpochs <= pot.FilledEpochs {
```

Comment:
We need a way to keep a perpetual pot.  Such a pot should distribute it's full balance every epoch, but the lock shouldn't disappear afterwards.  This is what OSMO rewards will probably go into.

Like the weekly minted amount should go into a perpetual pot that distributes for 1 epoch.  But new coins will be sent to this same pot every week until governance decides to vote to change it.

Reply:
I think we should file this as an issue, and make it in a second PR after this one gets merged

Reply:
I will raise a issue ticket for this.

Reply:
raised ticket. https://github.com/osmosis-labs/osmosis/issues/79